File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpaceRuby/Loot.rb
require_relative 'LootToUI'

module Deepspace

    class Loot
        def initialize (nSupplies, nWeapons, nShields, nHangars, nMedals, efficient=false, spaceCity=false)
            @nSupplies = nSupplies
            @nWeapons = nWeapons
            @nShields = nShields
            @nHangars = nHangars
            @nMedals = nMedals
            @efficient = efficient
            @spaceCity = spaceCity
        end

        def getUIversion
           return LootToUI.new(self) 
        end

        def nSupplies
            @nSupplies
        end

        def nWeapons
            @nWeapons
        end

        def nShields
            @nShields
        end

        def nHangars
            @nHangars
        end

        def nMedals
            @nMedals
        end

        def efficient
            @efficient
        end

        def spaceCity
            @spaceCity
        end

        def to_s
            return getUIversion.to_s
        end 
    end

end

File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpace_4/src/deepspace/Loot.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author manbolq
 */
public class Loot {
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    private boolean getEfficient;
    private boolean spaceCity;
    
    Loot(int nSupplies, int nWeapons, int nShields, int nHangars, int nMedals){
        this.nSupplies = nSupplies;
        this.nWeapons = nWeapons;
        this.nShields = nShields;
        this.nHangars = nHangars;
        this.nMedals = nMedals;
        getEfficient = false;
        spaceCity = false;
    }
    
     Loot(int nSupplies, int nWeapons, int nShields, int nHangars, int nMedals, boolean ef, boolean city){
        this.nSupplies = nSupplies;
        this.nWeapons = nWeapons;
        this.nShields = nShields;
        this.nHangars = nHangars;
        this.nMedals = nMedals;
        getEfficient = ef;
        spaceCity = city;
    }
    
    int getNSupplies(){
        return nSupplies;
    }
    
    int getNWeapons(){
        return nWeapons;
    }
    
    int getNShields(){
        return nShields;
    }
    
    int getNHangars(){
        return nHangars;
    }
    
    int getNMedals(){
        return nMedals;
    }
    
    public boolean getEfficient(){
        return getEfficient;
    }
    
    public boolean spaceCity(){
        return spaceCity;
    }
    
    @Override
    public String toString(){
        return getUIversion().toString();
    }
    
    
    LootToUI getUIversion(){
        return new LootToUI(this);
    }
}
